bindings/libdnf5/conf.i

```diff
@@ -105,3 +105,38 @@
 %include "libdnf/conf/config.hpp"
 %include "libdnf/conf/config_main.hpp"
 
+// The following adds shortcuts in Python for getting or setting
+// the configuration options using the configuration class attributes.
+//
+// To access all methods from the option, use standard getters as in C++ API.
+//
+// Example with the 'installroot':
+//
+// base = libdnf5.base.Base()
+// config = base.get_config()
+// config.installroot = '/tmp/installroot'
+// config.get_installroot_option().test('relative/path')
+//
+#if defined(SWIGPYTHON)
+%pythoncode %{
+import re
+
+def _config_option_getter(config_object, option_name):
+    return getattr(config_object, option_name)().get_value()
+
+def _config_option_setter(config_object, option_name, value):
+    getattr(config_object, option_name)().set(value)
+
+def create_config_option_attributes(cls):
+    for attr in dir(cls):
+        option_getter_match = re.search(r'get_(\w+)_option', attr)
+        if option_getter_match:
+            option_name = option_getter_match.group(1)
+            setattr(cls, option_name, property(
+                lambda self, attr=attr: _config_option_getter(self, attr),
+                lambda self, value, attr=attr: _config_option_setter(self, attr, value)
+            ))
```

Comment:
This would be of course better, but SWIG generates the wrappers for `ConfigMain` and `ConfigRepo` automatically and I am adding the attributes **after** the class is defined. Therefore I am using this approach. But of course if you know how to do it in a better way in SWIG I would be glad for improving it.

Reply:
I think you would also need to call something like `super().__getattr__` in the case of non `get_$1_option` named methods, otherwise those would be discarded by your implementation, right? And we still need to provide also the `__setattr__` implementation.

But I have no strong preference for my implementation, so if there are more advantages and supporting votes for your implementation I can switch :slightly_smiling_face:

Reply:
In other words, if I understand it right, your point is being that you prefer to redefine the Python class function responsible for handling the attributes instead of actually adding new attributes to this class (i.e. my version)...

Reply:
I just brought it up as an alternative way of doing this stuff (and this way is quite common in python world).
But I'm not gonna insist on it, especially when your solution with real properties has advantage of better attributes discoverability (which I did not realize when writing the first comment).
Now I'd say I like your solution more :)